tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -762,17 +768,17 @@ void TangoRosNode::StartPublishing() {
       std::thread(&TangoRosNode::PublishFisheyeImage, this);
   color_image_thread_.publish_thread =
       std::thread(&TangoRosNode::PublishColorImage, this);
-  mesh_marker_thread_.publish_thread =
-      std::thread(&TangoRosNode::PublishMeshMarker, this);
+  mesh_thread_.publish_thread =
+      std::thread(&TangoRosNode::PublishMesh, this);
   ros_spin_thread_ = std::thread(&TangoRosNode::RunRosSpin, this);
 }
 
 void TangoRosNode::StopPublishing() {
   if (run_threads_) {
     run_threads_ = false;
     if (device_pose_thread_.publish_thread.joinable()) {
-      if (!tango_data_available_ || !publish_pose_on_tf_
-          || !publish_pose_on_topic_) {
+      if (!tango_data_available_ || (!publish_pose_on_tf_
+          && !publish_pose_on_topic_)) {
```

Comment:
I guess this will need to be cleaned up when merged with master as the publish_pose_on_topic will be gone?